zb-classifier.js

```diff
@@ -1987,161 +2017,177 @@ class ZigbeeClassifier {
     this.addOnProperty(node, genLevelCtrlEndpoint);
   }
 
-  initOnOffButton(node, genOnOffOutputEndpoint) {
-    if (node.modelId.includes('motion')) {
-      // The IKEA Motion sensor has a modelId of 'TRADFRI motion sensor'
-      node.name = `${node.id}-motion`;
-      node['@type'] = ['MotionSensor'];
+  initOnOffButton(node, genOnOffOutputEndpoints) {
+    for (const idx in genOnOffOutputEndpoints) {
+      console.log('Processing endpoint', idx, '=',
+                  genOnOffOutputEndpoints[idx]);
+      const suffix = (idx === 0) ? '' : `${idx}`;
+      const endpoint = genOnOffOutputEndpoints[idx];
 
-      this.addButtonMotionProperty(node, genOnOffOutputEndpoint);
-      this.addEvents(node, {
-        motion: {
-          '@type': 'MotionEvent',
-          description: 'Motion detected',
-        },
-        'no-motion': {
-          '@type': 'MotionEvent',
-          description: 'Motion timeout',
-        },
-      });
-    } else {
-      node.name = `${node.id}-button`;
-      node['@type'] = ['PushButton'];
+      if (node.modelId.includes('motion')) {
+        // The IKEA Motion sensor has a modelId of 'TRADFRI motion sensor'
+        node.name = `${node.id}-motion`;
+        node['@type'] = ['MotionSensor'];
 
-      const onOffProperty =
-        this.addButtonOnProperty(node, genOnOffOutputEndpoint);
+        this.addButtonMotionProperty(node, endpoint, suffix);
+        this.addEvents(node, {
+          motion: {
+            '@type': 'MotionEvent',
+            description: 'Motion detected',
+          },
+          'no-motion': {
+            '@type': 'MotionEvent',
+            description: 'Motion timeout',
+          },
+        });
+      } else {
+        node.name = `${node.id}-button`;
+        node['@type'] = ['PushButton'];
+        const onOffProperty = this.addButtonOnProperty(node, endpoint, suffix);
 
-      onOffProperty.buttonIndex = 1;
+        onOffProperty.buttonIndex = 1;
 
-      this.addEvents(node, {
-        '1-pressed': {
-          '@type': 'PressedEvent',
-          description: 'On button pressed and released',
-        },
-      });
+        this.addEvents(node, {
+          '1-pressed': {
+            '@type': 'PressedEvent',
+            description: 'On button pressed and released',
+          },
+        });
+      }
     }
   }
 
-  initMultiLevelButton(node, genLevelCtrlOutputEndpoint,
-                       genOnOffOutputEndpoint) {
-    if (node.modelId.includes('motion') && genOnOffOutputEndpoint) {
+  initMultiLevelButton(node, genLevelCtrlOutputEndpoints,
```

Comment:
Should this be `initMultiLevelButtons` now?

Reply:
renames